File: 0318/ex03.js
// # <- Person 클래스 내부에서만 사용 가능
// # 외부에서는 접근 불가 바로 에러처리
// 캡슐화 정보은닉 es2022
class Person{
    #age;//접근이 허용되지 않음
    constructor(name,age){
        this.name = name;
        this.#age = age;
    }
    sayHi(){
        console.log(`this.name ${this.name} this.#age ${this.#age}`);
    }
}

const me = new Person("Lee",30);
me.sayHi();
console.log(me.name);
//console.log(me.#age); 접근불가 에러발생 클래스내부에서 닫혔다. private

File: self_c/ex0416/ex03.c
#define _CRT_SECURE_NOWARNINGS
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

typedef struct _Person{
    char name[20];
    int age;
    char address[100];
}Person;

int main(){
    Person p1 = {.name="홍길동", .age=25, .address="서울시 용산구 한남동"};
    
    printf("이름 : %s\n",p1.name);
    printf("나이 : %d\n",p1.age);
    printf("주소 : %s\n",p1.address);
    
    Person p2 = {"고길동",30,"서울시 용산구 한남동"};
    printf("이름 : %s\n",p2.name);
    printf("나이 : %d\n",p2.age);
    printf("주소 : %s\n",p2.address);
    
    // typedef로 별칭 달지 않았을 경우 struct _Person의 사이즈
    Person *p3 = malloc(sizeof(Person));
    
    strcpy(p3->name,"차길동");
    p3->age = 30;
    strcpy(p3 -> address,"서울시 강남구 논현동");
    printf("이름 : %s\n",p3->name);
    printf("나이 : %d\n",p3->age);
    printf("주소 : %s\n",p3->address);
    printf("이름 : %s\n",(*p3).name);
    printf("나이 : %d\n",(*p3).age);
    printf("주소 : %s\n",(*p3).address);
    free(p3);
    return 0;
}
